matrix-sdk/src/androidTest/java/org/matrix/androidsdk/AutoDiscoveryTest.kt

```diff
@@ -0,0 +1,575 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.androidsdk
+
+import android.os.Looper
+import okhttp3.OkHttpClient
+import org.junit.Assert
+import org.junit.FixMethodOrder
+import org.junit.Test
+import org.junit.runners.MethodSorters
+import org.matrix.androidsdk.common.MockOkHttpInterceptor
+import org.matrix.androidsdk.common.TestApiCallback
+import org.matrix.androidsdk.common.TestConstants
+import org.matrix.androidsdk.login.AutoDiscovery
+import java.util.concurrent.CountDownLatch
+
+
+@FixMethodOrder(MethodSorters.JVM)
+class AutoDiscoveryTest {
+
+    @Test
+    //If the returned status code is 404, then IGNORE.
```

Comment:
I prefer to see the comment above the @Test annotation (applicable to all tests)

Reply:
done